src/main/java/uk/gov/hmcts/ccd/domain/service/processor/DateTimeEntryProcessor.java

```diff
@@ -0,0 +1,83 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.domain.model.aggregated.CaseViewFieldBuilder;
+import uk.gov.hmcts.ccd.domain.model.aggregated.CommonField;
+import uk.gov.hmcts.ccd.domain.types.BaseType;
+import uk.gov.hmcts.ccd.domain.types.CollectionValidator;
+import uk.gov.hmcts.ccd.endpoint.exceptions.DataProcessingException;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static uk.gov.hmcts.ccd.domain.model.definition.FieldType.*;
+import static uk.gov.hmcts.ccd.domain.service.processor.DisplayContextParameter.*;
+
+@Component
+public class DateTimeEntryProcessor extends CaseDataFieldProcessor {
+
+    private static final List<String> SUPPORTED_TYPES = Arrays.asList(DATETIME, DATE);
+
+    private final DateTimeFormatParser dateTimeFormatParser;
+
+    @Autowired
+    public DateTimeEntryProcessor(CaseViewFieldBuilder caseViewFieldBuilder,
+                                  DateTimeFormatParser dateTimeFormatParser) {
+        super(caseViewFieldBuilder);
+        this.dateTimeFormatParser = dateTimeFormatParser;
+    }
+
+    @Override
+    protected JsonNode executeSimple(JsonNode node, CommonField field, BaseType baseType, String fieldPath) {
+        return !isNullOrEmpty(node)
+            && hasDisplayContextParameterType(field.getDisplayContextParameter(), DisplayContextParameterType.DATETIMEENTRY)
+            && isSupportedBaseType(baseType, SUPPORTED_TYPES) ?
+            createNode(field.getDisplayContextParameter(), node.asText(), baseType, fieldPath) :
+            node;
+    }
+
+    @Override
+    protected JsonNode executeCollection(JsonNode collectionNode, CommonField caseViewField, String fieldPath) {
+        final BaseType collectionFieldType = BaseType.get(caseViewField.getFieldType().getCollectionFieldType().getType());
+
+        if (hasDisplayContextParameterType(caseViewField.getDisplayContextParameter(), DisplayContextParameterType.DATETIMEENTRY)
+            && isSupportedBaseType(collectionFieldType, SUPPORTED_TYPES)) {
+            ArrayNode newNode = MAPPER.createArrayNode();
+            collectionNode.forEach(item -> {
+                JsonNode newItem = item.deepCopy();
+                ((ObjectNode)newItem).replace(CollectionValidator.VALUE,
+                    createNode(caseViewField.getDisplayContextParameter(), item.get(CollectionValidator.VALUE).asText(), collectionFieldType, fieldPath));
+                newNode.add(newItem);
+            });
+
+            return newNode;
+        }
+
+        return collectionNode;
+    }
+
+    private TextNode createNode(String displayContextParameter, String valueToConvert, BaseType baseType, String fieldPath) {
+        String format = getDisplayContextParameterOfType(displayContextParameter, DisplayContextParameterType.DATETIMEENTRY)
+            .map(DisplayContextParameter::getValue)
+            .orElse("");
```

Comment:
I am not sure about the  requirements . Can we create a "" format ?  Would that be default ? or it will lead directly in exception.   If we know before it we can treat it  as  throw new DataProcessingException() . in Common  error handling place.